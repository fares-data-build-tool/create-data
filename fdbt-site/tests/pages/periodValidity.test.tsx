/* eslint-disable global-require */

import * as React from 'react';
import { shallow } from 'enzyme';
import PeriodValidity from '../../src/pages/periodValidity';

describe('pages', () => {
    describe('periodValidity', () => {
        it('should render correctly', () => {
            const tree = shallow(<PeriodValidity errors={[]} />);
            expect(tree).toMatchSnapshot();
        });

        it('should render error messaging when errors are passed', () => {
            const tree = shallow(
                <PeriodValidity
                    errors={[
                        {
                            errorMessage: 'Choose an option regarding your period ticket validity',
                            id: 'period-validity-error',
                        },
                    ]}
                />,
            );
            expect(tree).toMatchSnapshot();
        });
    });
});
