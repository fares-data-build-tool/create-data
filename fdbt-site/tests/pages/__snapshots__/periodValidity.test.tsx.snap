// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages periodValidity should render correctly 1`] = `
<Component
  description="Period Validity selection page of the Create Fares Data Service"
  errors={Array []}
  title="Period Validity - Create Fares Data Service"
>
  <CsrfForm
    action="/api/periodValidity"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="period-validity-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="period-validity-page-heading"
          >
            When does the product expire?
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="period-validity-hint"
        >
          We need to know the time that this product would be valid until
        </span>
        <RadioConditionalInput
          fieldset={
            Object {
              "heading": Object {
                "content": Any<String>,
                "hidden": true,
                "id": "period-validity",
              },
              "radioError": Array [],
              "radios": Array [
                Object {
                  "id": "period-end-calendar",
                  "label": " At the end of a calendar day",
                  "name": "periodValid",
                  "radioButtonHint": Object {
                    "content": "For example, a ticket purchased at 3pm would be valid until midnight on its day of expiry",
                    "id": "period-end-calendar-hint",
                  },
                  "value": "endOfCalendarDay",
                },
                Object {
                  "id": "period-twenty-four-hours",
                  "label": "At the end of a 24 hour period from purchase",
                  "name": "periodValid",
                  "radioButtonHint": Object {
                    "content": "For example, a ticket purchased at 3pm will be valid until 3pm on its day of expiry",
                    "id": "period-twenty-four-hours-hint",
                  },
                  "value": "24hr",
                },
                Object {
                  "dataAriaControls": "period-validity-end-of-service-required-conditional",
                  "id": "period-end-of-service",
                  "inputErrors": Array [],
                  "inputHint": Object {
                    "content": "Enter an end time for your service day",
                    "hidden": true,
                    "id": "end-of-service-day-hint",
                  },
                  "inputType": "text",
                  "inputs": Array [
                    Object {
                      "defaultValue": "",
                      "id": "service-end-time",
                      "label": "End time",
                      "name": "serviceEndTime",
                    },
                  ],
                  "label": "End of service day",
                  "name": "periodValid",
                  "radioButtonHint": Object {
                    "content": "For example, a ticket purchased at 3pm would be valid until the end of your service day on its day of expiry",
                    "id": "period-end-of-service-hint",
                  },
                  "value": "endOfServiceDay",
                },
              ],
            }
          }
          key="period-validity"
        />
      </fieldset>
    </div>
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;

exports[`pages periodValidity should render error messaging when errors are passed 1`] = `
<Component
  description="Period Validity selection page of the Create Fares Data Service"
  errors={
    Array [
      Object {
        "errorMessage": "Choose an option regarding your period ticket validity",
        "id": "period-validity-error",
      },
    ]
  }
  title="Period Validity - Create Fares Data Service"
>
  <CsrfForm
    action="/api/periodValidity"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={
        Array [
          Object {
            "errorMessage": "Choose an option regarding your period ticket validity",
            "id": "period-validity-error",
          },
        ]
      }
    />
    <div
      className="govuk-form-group govuk-form-group--error"
    >
      <fieldset
        aria-describedby="period-validity-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="period-validity-page-heading"
          >
            When does the product expire?
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="period-validity-hint"
        >
          We need to know the time that this product would be valid until
        </span>
        <RadioConditionalInput
          fieldset={
            Object {
              "heading": Object {
                "content": Any<String>,
                "hidden": true,
                "id": "period-validity",
              },
              "radioError": Array [],
              "radios": Array [
                Object {
                  "id": "period-end-calendar",
                  "label": " At the end of a calendar day",
                  "name": "periodValid",
                  "radioButtonHint": Object {
                    "content": "For example, a ticket purchased at 3pm would be valid until midnight on its day of expiry",
                    "id": "period-end-calendar-hint",
                  },
                  "value": "endOfCalendarDay",
                },
                Object {
                  "id": "period-twenty-four-hours",
                  "label": "At the end of a 24 hour period from purchase",
                  "name": "periodValid",
                  "radioButtonHint": Object {
                    "content": "For example, a ticket purchased at 3pm will be valid until 3pm on its day of expiry",
                    "id": "period-twenty-four-hours-hint",
                  },
                  "value": "24hr",
                },
                Object {
                  "dataAriaControls": "period-validity-end-of-service-required-conditional",
                  "id": "period-end-of-service",
                  "inputErrors": Array [],
                  "inputHint": Object {
                    "content": "Enter an end time for your service day",
                    "hidden": true,
                    "id": "end-of-service-day-hint",
                  },
                  "inputType": "text",
                  "inputs": Array [
                    Object {
                      "defaultValue": "",
                      "id": "service-end-time",
                      "label": "End time",
                      "name": "serviceEndTime",
                    },
                  ],
                  "label": "End of service day",
                  "name": "periodValid",
                  "radioButtonHint": Object {
                    "content": "For example, a ticket purchased at 3pm would be valid until the end of your service day on its day of expiry",
                    "id": "period-end-of-service-hint",
                  },
                  "value": "endOfServiceDay",
                },
              ],
            }
          }
          key="period-validity"
        />
      </fieldset>
    </div>
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;
